hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/server/HermesServer.java

```diff
@@ -97,10 +96,19 @@ public void start() {
         if (configFactory.getBooleanProperty(FRONTEND_TOPIC_METADATA_REFRESH_JOB_ENABLED)) {
             topicMetadataLoadingJob.start();
         }
+        healthCheckService.startup();
         readinessChecker.start();
     }
 
-    public void gracefulShutdown() throws InterruptedException {
+    public void stop() throws InterruptedException {
+        boolean isGraceful = configFactory.getBooleanProperty(FRONTEND_GRACEFUL_SHUTDOWN_ENABLED);
+        if(isGraceful) {
+            gracefullyPrepareForShutdown();
+        }
+        shutdown();
+    }
+
+    public void gracefullyPrepareForShutdown() throws InterruptedException {
```

Comment:
In my opinion, this method prepares the hermes-frontend instance for a graceful shutdown, so probably its name should be: `prepareForGracefulShutdown`. WDYT?